registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/helper/RestHelperImpl.java

```diff
@@ -0,0 +1,149 @@
+import io.mosip.registration.processor.stages.exception.RestServiceException;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import lombok.NoArgsConstructor;
+import reactor.core.publisher.Mono;
+
+@Component
+@NoArgsConstructor
+public class RestHelperImpl implements RestHelper {
+
+	private static Logger mosipLogger = RegProcessorLogger.getLogger(RestHelperImpl.class);
+
+	private String authToken;
+	@Autowired
+	private Environment env;
+	@Autowired
+	private ObjectMapper mapper;
+
+	@Override
+	public Supplier<Object> requestAsync(@Valid RestRequestDTO request) {
+		try {
+			Mono<?> sendRequest = request(request, getSslContext());
+			sendRequest.subscribe();
+			return () -> sendRequest.block();
+		} catch (RestServiceException e) {
+			System.out.println(ExceptionUtils.getStackTrace(e));
+			return () -> new RestServiceException("UNABLE_TO_PROCESS", e);
+		}
+	}
+
+	private SslContext getSslContext() throws RestServiceException {
+		try {
+			return SslContextBuilder.forClient().trustManager(InsecureTrustManagerFactory.INSTANCE).build();
+		} catch (SSLException e) {
+			System.out.println(ExceptionUtils.getStackTrace(e));
+			throw new RestServiceException("UNKNOWN_ERROR", e);
+		}
+	}
+
+	private Mono<?> request(RestRequestDTO request, SslContext sslContext) {
+		WebClient webClient;
+		Mono<?> monoResponse;
+		RequestBodySpec uri;
+		ResponseSpec exchange;
+		RequestBodyUriSpec method;
+
+		if (request.getHeaders() != null) {
+			webClient = WebClient.builder()
+					.clientConnector(new ReactorClientHttpConnector(builder -> builder.sslContext(sslContext)))
+					.baseUrl(request.getUri())
+					.defaultHeader(HttpHeaders.CONTENT_TYPE, request.getHeaders().getContentType().toString()).build();
+		} else {
+			webClient = WebClient.builder()
+					.clientConnector(new ReactorClientHttpConnector(builder -> builder.sslContext(sslContext)))
+					.baseUrl(request.getUri()).build();
+		}
+
+		method = webClient.method(request.getHttpMethod());
+		if (request.getParams() != null && request.getPathVariables() == null) {
+			uri = method.uri(builder -> builder.queryParams(request.getParams()).build());
+		} else if (request.getParams() == null && request.getPathVariables() != null) {
+			uri = method.uri(builder -> builder.build(request.getPathVariables()));
+		} else {
+			uri = method.uri(builder -> builder.build());
+		}
+
+		uri.cookie("Authorization", getAuthToken());
+
+		if (request.getRequestBody() != null) {
+			exchange = uri.syncBody(request.getRequestBody()).retrieve();
+		} else {
+			exchange = uri.retrieve();
+		}
+
+		monoResponse = exchange.bodyToMono(request.getResponseType());
+
+		return monoResponse;
+	}
+
+	private String getAuthToken() {
+		if (EmptyCheckUtils.isNullEmpty(authToken)) {
+			generateAuthToken();
+			return authToken;
+		} else {
+			return authToken;
+		}
+	}
+
+	private void generateAuthToken() {
```

Comment:
we have to reuse same auth token for multiple requests. need to change this method